src/State/Provider/ReadProvider.php

```diff
@@ -84,6 +88,8 @@ public function provide(Operation $operation, array $uriVariables = [], array $c
                 || ($operation instanceof Put && !($operation->getAllowCreate() ?? false))
             )
         ) {
+            // In case the dev just forgot to implement it
+            $this->logger?->debug('No provider registered for {resource_class}', ['resource_class' => $resourceClass]);
```

Comment:
imo this line should be at line :81

Reply:
I wasn't sure that was something useful to report for PUT and POST, but that's easy enough to change.